test/end-to-end/core.sh

```diff
@@ -137,6 +137,11 @@ oc login -u e2e-user
 oc project test
 oc whoami
 
+echo "[INFO] Running a CLI command in a container using the service account"
+out=$(oc run cli-with-token --attach --image=openshift/origin:${TAG} --restart=Never -- cli status --loglevel=4 2>&1)
```

Comment:
run `cli whoami` and verify the identity matches?
add the service account to a role that can list pods and do a `kube get pods` to make sure kubectl paths work as well?